batch/snippets/src/test/java/BatchTemplateIT.java

```diff
@@ -0,0 +1,210 @@
+import com.google.cloud.compute.v1.InstanceProperties;
+import com.google.cloud.compute.v1.InstanceTemplate;
+import com.google.cloud.compute.v1.InstanceTemplatesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Scheduling;
+import com.google.cloud.compute.v1.Scheduling.OnHostMaintenance;
+import com.google.cloud.compute.v1.Scheduling.ProvisioningModel;
+import com.google.cloud.compute.v1.ServiceAccount;
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BatchTemplateIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String REGION = "europe-north1";
+  private static String PROJECT_NUMBER;
+  private static String SCRIPT_JOB_NAME;
+  private static InstanceTemplate INSTANCE_TEMPLATE;
+
+  private ByteArrayOutputStream stdOut;
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp()
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    // Get project number from project id.
+    try (ProjectsClient projectsClient = ProjectsClient.create()) {
+      PROJECT_NUMBER = projectsClient.getProject(String.format("projects/%s", PROJECT_ID)).getName()
+          .split("/")[1];
+    }
+
+    String uuid = String.valueOf(UUID.randomUUID());
+    SCRIPT_JOB_NAME = "test-job-template-" + uuid;
+    INSTANCE_TEMPLATE = createInstanceTemplate();
+    TimeUnit.SECONDS.sleep(10);
+
+    stdOut.close();
+    System.setOut(out);
+  }
+
+  @AfterClass
+  public static void cleanup()
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+
+    deleteInstanceTemplate();
+    DeleteJob.deleteJob(PROJECT_ID, REGION, SCRIPT_JOB_NAME);
```

Comment:
Do we know that both of these will get executed if there is a crash?

Reply:
If the deleteJob doesn't get executed, it will deleted when the test runs the next time (added code for deleting stale resources).